src/Formik.tsx

```diff
@@ -1001,7 +1001,11 @@ function prepareDataForValidation<T extends FormikValues>(
             return value !== '' ? value : undefined;
           }
         });
-      } else if (typeof values[key] === 'object' && values[key] !== null) {
+      } else if (
+        typeof values[key] === 'object' &&
```

Comment:
Can you add a comment about this? also can we use `isObject()`?

Reply:
Mmm, I'm not sure if possible to use isObject. Using 

> values[key] instanceof Object

 also returns true...

I didn't find any other possible solution to avoid this.

Reply:
The problem is...
`{ date: Moment() /* moment instance */ }`

How we can check that the date property is an object?

Reply:
We can use [is-plain-object](https://github.com/jonschlinkert/is-plain-object) for this

Reply:
Yes, changed.